### PR TITLE
Use `thread-id` in thief_id call

### PR DESCRIPTION
**Pull request by hodgesds**
Hi, I'm new and wanted to help out a little if I can. I might not have followed all the conversation in the related issue but happy to help out if I can.

Resolves: #335

### CODE REVIEW

File: src/iter/internal.rs [hunk 1]
```diff
@@ -147,8 +148,7 @@ impl Splitter {
     fn thief_id() -> usize {
         // The actual `ID` value is irrelevant.  We're just using its TLS
         // address as a unique thread key, faster than a real thread-id call.
-        thread_local!{ static ID: bool = false }
-        ID.with(|id| id as *const bool as usize)
+        thread_id::get()
```
Comment (cuviper) [inline src/iter/internal.rs]: The comment should be updated too, like:

    // The actual `ID` value is irrelevant, as long as it's unique.

**Comment by hodgesds** [comment]
Thanks for the comments, I'll try to take a look at doing some performance testing tonight.

**Comment by nikomatsakis** [comment]
As it happens, I just made [a wiki page](https://github.com/nikomatsakis/rayon/wiki/Benchmarking-Rayon) about benchmarking that you may find useful. I can try to do some measurements too.

**Comment by nikomatsakis** [comment]
This is superceded by #354, right? I think I prefer that approach.

**Comment by nikomatsakis** [comment]
(To be clear, I appreciate the PR @hodgesds -- just seems better if we can avoid the dependency and make things a bit cleaner in the bargain. =)

Sorry for the long delay btw.

**Pull request closed without merging**